File: XALDAPP/corefinanciero/src/main/java/com/proyecto/xald/corefinanciero/PayloadSincronizacionDTO.kt
package com.proyecto.xald.corefinanciero

/**
 * Objeto de solo lectura empaquetado por el núcleo financiero.
 * Sirve de contrato seguro para enviar datos al módulo :syncqueue sin exponer la tabla original.
 */
data class PayloadSincronizacionDTO(
    val idTransaccion: String,
    val monto: Double,
    val comercio: String,
    val fechaTimestamp: Long
)

File: XALDAPP_FLU/packages/syncqueue/lib/src/dtos/payload_sincronizacion_dto.dart
class PayloadSincronizacionDTO {
  final String idTransaccion;
  final double monto;
  final String comercio;
  final String categoria;
  final String fechaIso;

  const PayloadSincronizacionDTO({
    required this.idTransaccion,
    required this.monto,
    required this.comercio,
    required this.categoria,
    required this.fechaIso,
  });

  Map<String, dynamic> toJson() => {
    'id_transaccion': idTransaccion,
    'monto': monto,
    'comercio': comercio,
    'categoria': categoria,
    'fecha_iso': fechaIso,
  };
}